### ISSUE TITLE
No mic input during meetings

### ISSUE THREAD

**Issue by rdlittle** [IsmaelMartinez/teams-for-linux#277]
No audio input during Teams meeting.  Even though the mic is functional for other applications (Skype, for example).  I am using a Koss USB headset.

**Steps to reproduce the behavior:**
1. Go to 'Calendar'
2. Click on 'Join' button in the event
3. Ensure that mic is not muted (i.e. mic icon is not crossed out)
4. Join the meeting

**Expected behavior**
Participants should hear me when I speak.

**System**
- Desktop: Fedora 30, KDE spin
 - OS: Linux 5.2.11-200.fc30.x86_64
 - Installation package rpm
 - Version: 0.7.0

I have _sometimes_ been able to achieve success by:
1. Opening the system's audio settings
2. Switch the default input from the USB device to some other input - typically the webcam
3. Apply the change
4. Reopen the system's audio settings
5. Change the default input back to the USB headset
6. Restart Teams for Linux

Repeat the above once or twice more.

I highly suspect this is more distro related than it is app related.  Nevertheless I would appreciate any advice you can offer on how I can troubleshoot the problem more effectively.

**Comment by IsmaelMartinez** [comment]
Hi @rdlittle, someone in the past had issues with the mic and, by reconfiguring the pulse audio, it got solved (not sure if that applies to fedora)

Another thing that I might suspect is if the "teams-for-linux" client is trying to get a lock of the mic and can't because there is something already there running.

Can you run the app with:
```bash
teams-for-linux --try-supported-channel-layouts
```

If the command works, it means is the same issue as #275.

Hope helps.

**Comment by rdlittle** [comment]
Hello Ismael
The command line argument didn't do much for me.  I did notice this in the output though - Don't know if its relevant:

> [25305:0906/154517.084758:ERROR:adm_helpers.cc(73)] Failed to query stereo recording.

**Comment by IsmaelMartinez** [comment]
Will it be possible for you to try other desktop instead of KDE? Doing a bit of a google there are a few pointers or issues with KDE and that issue. No worth putting links here as I don't think are related and always point to system reconfiguration. It looks like it is an OS config. Did you try re-configuring the pulse audio?

**Comment by rdlittle** [comment]
I _may_ have it figured out thanks to things you and @bubble75 have said.  You pointed out the pulseaudio config which put me in mind of an issue that used to occur with Skype a few years ago.   At that time, Skype for Linux did not "see" the mic unless the Pulse Audio Volume Control app was up and running.  @bubble75 mentioned voicemail configuration - which I didn't even know existed - and MS sure doesn't make it easy to find.
So I started PA control running.  Then started Teams and went to the Settings menu where I saw an option to change voicemail greeting:
![MicrosoftTeams-image](https://user-images.githubusercontent.com/3808330/64475837-d22b7700-d155-11e9-804d-c1f6b8099220.png)
From there I discovered an option to "Show device settings"
![Screenshot_20190907_094209](https://user-images.githubusercontent.com/3808330/64475857-069f3300-d156-11e9-9fd4-001fcb67d75f.png)
Finally, I was presented with a dialog box which showed my current devices
![Screenshot_20190907_094134](https://user-images.githubusercontent.com/3808330/64475874-47974780-d156-11e9-8116-da75a01be310.png)
As suspected, it had the wrong device under "Microphone" even though PA Vol Ctrl showed otherwise.  So I changed it to my USB headset and suddenly I had audio input.

Why MS buried this config setting so deeply is quite beyond me.

I'll know more on Monday when I try to join a meeting.

**Comment by IsmaelMartinez** [comment]
hi @rdlittle, Let us know how you get on. It looks like a configuration issue rather than a bug. It will be great if you can add something to the know issues section if setting the configuration work. Cheers

**Comment by rdlittle** [comment]
Hi @IsmaelMartinez 

Having started Teams For Linux with "--try-supported-channel-layouts" passed to the app, I was finally able to connect to a meeting with full audio both input and output.

I've discovered though, that _without_ the --try-supported-channel-layouts command line arg, the 'Calls' section in 'Settings' does not show up.  As a result, there's no way to get to the device settings:

**Without extra arg:**
![Screenshot_20190909_104543](https://user-images.githubusercontent.com/3808330/64541165-6aa92f00-d2ef-11e9-8ee4-3f8ebfc2c455.png)

If I start the app with --try-supported-channel-layouts arg, the 'Calls' section _does_ show up:

**With extra arg:**
![Screenshot_20190909_104630](https://user-images.githubusercontent.com/3808330/64541439-d9868800-d2ef-11e9-8d46-952db4e30267.png)

Interestingly enough, the phone icon will be enabled in the Chat area either way.  
I connected to the meeting with the command line args in place, but I am reluctant to try without.  I can't afford the risk  :)

**Comment by bubble75** [comment]
At today's meeting my colleagues told me that I have been heard with a poor quality (with that commandline arg), which became worse even more within a few minutes and it improved only after I reconnected from android phone with the same wired headset. I'll try to make some test calls tomorrow, trying to determine what can improve sound quality on a notebook (to strike-out app itself from the list of suspects).

**Comment by IsmaelMartinez** [comment]
Keep us updated. Hopefully we can help but it can be a system configuration.

**Comment by IsmaelMartinez** [comment]
I am closing this as this was about not having a mic, not the quality of it. Open another issue if you need further help.
